### PR TITLE
Mesos interaction

### PR DESCRIPTION
**Pull request by jerith**

### CODE REVIEW

File: mesosclient/testing/fake_mesos_test.go [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+	ts.Equal(resp.StatusCode, 405)
+}
+
+// Non-protobuf requests return 415.
+func (ts *FakeMesosTests) Test_API_bad_content_type() {
+	fm := NewFakeMesos()
+	ts.AddCleanup(fm.Close)
+
+	resp := ts.getResp(http.Post(fm.GetAPIURL(), "image/png", strings.NewReader("not a png")))
+	// resp := ts.postAPI("http://10.0.91.11:5050/api/v1", "image/png", strings.NewReader("not a png"))
```
Comment (bearnard) [inline mesosclient/testing/fake_mesos_test.go]: intentional?

Comment (jerith) [reply mesosclient/testing/fake_mesos_test.go]: Nope, that shouldn't be there. Thanks.

File: mesosclient/mesos.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+// Client is a Mesos API client.
+type Client struct {
+	sender calls.Sender
+}
+
+// NewClient builds a new Client object that queries a Mesos API endpoint at
+// the given base URL. The baseURL parameter should have the form
+// "http://host:port".
+func NewClient(baseURL string) *Client {
+	endpoint := httpcli.Endpoint(fmt.Sprintf("%s/api/v1", baseURL))
```
Comment (JayH5) [inline mesosclient/mesos.go]: I'm wondering about other config options, like multiple URLs, timeouts, TLS, etc. But I guess that is something for another PR/when we care about it.

Comment (jerith) [reply mesosclient/mesos.go]: `mesos-go` doesn't really support that stuff (although we could probably use `context.Context` for timeouts) so we'd have to add it ourselves. I'd prefer to wait until we have a better idea of how and where we'll be calling Mesos before doing that.

**Pull request merged**